syft/grid/utils/autoscale/gcloud.py

```diff
@@ -34,23 +35,145 @@ def __init__(self, credentials, project_id, region):
+        )
+        self.config += image
+
+        node = terrascript.resource.google_compute_instance(
             name,
             name=name,
             machine_type=machine_type,
             zone=zone,
-            boot_disk={"initialize_params": {"image": image_family}},
+            boot_disk={"initialize_params": {"image": "${" + image.self_link + "}"}},
+            network_interface={
+                "network": "default",
+                "access_config": {"nat_ip": "${" + pygrid_network_ip.address + "}"},
+            },
+            metadata_startup_script="""
+                #!/bin/bash
+                apt-get update
+                apt-get -y upgrade
+                sudo -i bash -c 'pip install git+https://github.com/OpenMined/PyGridNetwork.git'
+                sudo -i bash -c 'echo Starting PyGridNetwork & \
+                python -m gridnetwork --port=80 --start_local_db'""",
+        )
+        self.config += node
+
+        with open("main.tf.json", "w") as main_config:
+            json.dump(self.config, main_config, indent=2, sort_keys=False)
+
+        if apply:
+            if IPython.get_ipython():
+                terraform_notebook.apply()
+            else:
+                terraform_script.apply()
+
+    def create_gridnode(self, name, machine_type, zone, gridnetwork_name=None, apply=True):
+        """
+        args:
+            name: name of the compute instance
+            machine_type: the type of machine
+            zone: zone of your GCP project
+            gridnetwork_name: name of gridnetwork instance created
+            apply: to call terraform apply at the end
+        """
+        if not gridnetwork_name:
+            self.expose_port(name="pygrid", ports=[80], apply=False)
+
+        with open("terraform.tfstate", "r") as out:
+            outputs = json.load(out)["outputs"]
+
+        gridnetwork_ip = outputs[gridnetwork_name + "-ip"]["value"]
+        pygrid_network_address = "--gateway_url=http://" + gridnetwork_ip if gridnetwork_ip else ""
+
+        image = terrascript.data.google_compute_image(
+            name + "pytorch",
+            family="pytorch-latest-gpu-debian-10",
+            project="deeplearning-platform-release",
+        )
+        self.config += image
+
+        node = terrascript.resource.google_compute_instance(
+            name,
+            name=name,
+            machine_type=machine_type,
+            zone=zone,
+            boot_disk={"initialize_params": {"image": "${" + image.self_link + "}"}},
             network_interface={"network": "default", "access_config": {}},
+            metadata_startup_script=f"""
+                #!/bin/bash
+                apt-get update
+                apt-get -y upgrade
+                sudo -i bash -c 'pip install notebook==5.7.8'
+                sudo -i bash -c 'pip install git+https://github.com/OpenMined/PyGridNode.git'
+                sudo -i bash -c 'echo Starting Node {name} \
```

Comment:
GridNode Repo is archived now, this should be updated to new method for creating nodes from Pygrid

Reply:
Now as the refactoring has completed, I will update it to the new methods shortly.